src/platform/packages/private/kbn-index-editor/src/components/grid_custom_renderers/column_header_popover.tsx

```diff
@@ -0,0 +1,255 @@
+
+export const ColumnHeaderPopover = ({
+  isColumnInEditMode,
+  setEditingColumnIndex,
+  isSavedColumn,
+  initialColumnName,
+  initialColumnType,
+  isUnsupportedESQLType,
+  columnIndex,
+  telemetryService,
+  originalColumnDisplay,
+}: PropsWithChildren<ColumnHeaderPopoverProps>) => {
+  const { euiTheme } = useEuiTheme();
+  const { columnType, setColumnType, columnName, setColumnName, saveColumn, validationError } =
+    useAddColumnName(initialColumnName, initialColumnType);
+
+  const canSubmit = columnType && columnName.length > 0 && !validationError;
+
+  const onSubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      event.stopPropagation();
+
+      if (columnName && !validationError) {
+        setEditingColumnIndex(null);
+        saveColumn();
+      } else {
+        telemetryService.trackEditInteraction({
+          actionType: 'add_column',
+          failureReason: validationError || 'EMPTY_NAME',
+        });
+      }
+    },
+    [columnName, validationError, setEditingColumnIndex, saveColumn, telemetryService]
+  );
+
+  const columnLabel = isPlaceholderColumn(initialColumnName) ? (
+    <FormattedMessage
+      id="indexEditor.flyout.grid.columnHeader.add"
+      defaultMessage="Add a column…"
+    />
+  ) : (
+    // The default column header display comming from UnifiedDataTable, the type icon + column name
+    <EuiFlexGroup alignItems="center" gutterSize="s" wrap={false} css={{ cursor: 'pointer' }}>
+      {isUnsupportedESQLType && (
+        <EuiIconTip
+          type="warning"
+          color="warning"
+          size="m"
+          content={i18n.translate('indexEditor.columnHeader.unsupportedWarning', {
+            defaultMessage:
+              '{unsupportedType} type has partial suport, editions you do will be saved but will not be visible in this editor after closing it.',
+            values: { unsupportedType: initialColumnType },
+          })}
+          className="fieldWarningTip"
+          anchorProps={{
+            css: { display: 'flex', marginLeft: euiTheme.size.xxs },
+          }}
+        />
+      )}
+      {originalColumnDisplay}
+    </EuiFlexGroup>
+  );
+
+  const errorMessage = useMemo(() => {
+    if (!validationError) return;
+    return errorMessages[validationError]
+      ? errorMessages[validationError](columnName)
+      : validationError;
+  }, [validationError, columnName]);
+
+  const returnFocus = useCallback(() => {
+    requestAnimationFrame(() => {
+      const headerWrapper = findElementBySelectorOrRef(`[${COLUMN_INDEX_PROP}="${columnIndex}"]`);
+
+      if (headerWrapper) {
+        headerWrapper.focus();
+      }
+    });
+
+    return false;
+  }, [columnIndex]);
+
+  if (isSavedColumn) {
+    return columnLabel;
+  }
+
+  const triggerButton = (
+    // This button is keyboard accesible via the column actions menu.
+    // eslint-disable-next-line @elastic/eui/accessible-interactive-element
```

Comment:
We are the first users of this lint. If you wrap it in a div with the tabIndex does it mess up with the accessibiliy?